sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -108,9 +108,9 @@
      */
     private static BiFunction<HttpClientResponse, Connection, Publisher<HttpResponse>> responseDelegate(
         final HttpRequest restRequest, final boolean disableBufferCopy) {
-        return (reactorNettyResponse, reactorNettyConnection) ->
-            Mono.just(new ReactorNettyHttpResponse(reactorNettyResponse, reactorNettyConnection, restRequest,
-                disableBufferCopy));
+        return (reactorNettyResponse, reactorNettyConnection) -> Mono.defer(() -> Mono.using(() ->
+            new ReactorNettyHttpResponse(reactorNettyResponse, reactorNettyConnection, restRequest, disableBufferCopy),
+            Mono::just, HttpResponse::close));
```

Comment:
As discussed this may result in early disposal of response before the inner response-body is fully consumed.

Reply:
Changed design pattern to use `Reference` instead of `Mono.using`.